### PR TITLE
3 make sauces controller

### LINKED ISSUE #3: Make sauces controller
Make a backend controller in which:
1. Admin user can add new sauce 
2. Admin user can remove sauce by id. When sauce is removed, all connection of this sauce to other kebabs should be removed (TypesOfSaucesInKebab table).
3. Admin user can edit sauce name by id
4. Get all sauces

### PR DESCRIPTION
**Pull request by MatWisie**

### CODE REVIEW

File: KebabBackend/app/Http/Controllers/Api/SauceTypeController.php [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+ *         @OA\Property(property="name", type="string", example="Sweet Chili")
+ *     }
+ * )
+ */
+
+class SauceTypeController extends Controller
+{
+    /**
+     * @OA\Get(
+     *     path="/api/sauce-types",
```
Comment (Wiktor-Wojciechowski) [inline KebabBackend/app/Http/Controllers/Api/SauceTypeController.php]:
```suggestion
     *     path="/api/saucetypes",
```

File: KebabBackend/app/Http/Controllers/Api/SauceTypeController.php [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+     */
+    public function index()
+    {
+        $sauceTypes = SauceType::all();
+        return response()->json($sauceTypes);
+    }
+
+    /**
+     * @OA\Post(
+     *     path="/api/sauce-types",
```
Comment (Wiktor-Wojciechowski) [inline KebabBackend/app/Http/Controllers/Api/SauceTypeController.php]:
```suggestion
     *     path="/api/saucetypes",
```

File: KebabBackend/app/Http/Controllers/Api/SauceTypeController.php [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+        return response()->json([
+            'message' => 'SauceType added successfully',
+            'saucetype' => $sauceType,
+        ], 200);
+    }
+
+    /**
+     * @OA\Put(
+     *     path="/api/sauce-types/{id}",
```
Comment (Wiktor-Wojciechowski) [inline KebabBackend/app/Http/Controllers/Api/SauceTypeController.php]:
```suggestion
     *     path="/api/saucetypes/{id}",
```

File: KebabBackend/app/Http/Controllers/Api/SauceTypeController.php [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+        return response()->json([
+            'message' => 'Sauce type updated successfully',
+            'sauceType' => $sauceType,
+        ]);
+    }
+
+    /**
+     * @OA\Delete(
+     *     path="/api/sauce-types/{id}",
```
Comment (Wiktor-Wojciechowski) [inline KebabBackend/app/Http/Controllers/Api/SauceTypeController.php]:
```suggestion
     *     path="/api/saucetypes/{id}",
```

File: KebabBackend/storage/api-docs/api-docs.json [hunk 5]
```diff
@@ -1055,6 +1055,159 @@
                 }
             }
         },
+        "/api/sauce-types": {
```
Comment (Wiktor-Wojciechowski) [inline KebabBackend/storage/api-docs/api-docs.json]:
```suggestion
        "/api/saucetypes": {
```

File: KebabBackend/storage/api-docs/api-docs.json [hunk 6]
```diff
@@ -1055,6 +1055,159 @@
...
+                                    },
+                                    "type": "object"
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "/api/sauce-types/{id}": {
```
Comment (Wiktor-Wojciechowski) [inline KebabBackend/storage/api-docs/api-docs.json]:
```suggestion
        "/api/saucetypes/{id}": {
```

**Review by Wiktor-Wojciechowski — CHANGES_REQUESTED** [review]
These paths have typos in them

File: KebabBackend/routes/api.php [hunk 7]
```diff
@@ -70,8 +71,18 @@
...
     Route::middleware(['auth:sanctum', 'admin'])->post('/', [MeatTypeController::class, 'store']);
 
-    Route::middleware(['auth:sanctum', 'admin'])->put('{kebab}', [MeatTypeController::class, 'update']);
+    Route::middleware(['auth:sanctum', 'admin'])->put('{id}', [MeatTypeController::class, 'update']);
 
-    Route::middleware(['auth:sanctum', 'admin'])->delete('{kebab}', [MeatTypeController::class, 'destroy']);
+    Route::middleware(['auth:sanctum', 'admin'])->delete('{id}', [MeatTypeController::class, 'destroy']);
+});
+
+Route::prefix('saucetypes')->group(function () {
```
Comment (W-Worach) [inline KebabBackend/routes/api.php:86]: > These paths have typos in them

I the typo is not in docs but in route 

```suggestion
Route::prefix('sauce-types')->group(function () {
```

Comment (Wiktor-Wojciechowski) [reply KebabBackend/routes/api.php:86]: This change would make the paths a bit inconsistent. Perhaps it is better to change them to saucetypes to fit together with meattypes. Or we could change meattypes into meat-types too

**Comment by MatWisie** [comment]
Alright alright, problem solved in https://github.com/MatWisie/KebabFinder-API/pull/34/commits/aed686ac870aaa896bd3ed48df7fce7cc7e59bf1

**Review by Wiktor-Wojciechowski — APPROVED** [review]

**Pull request merged**